app/jobs/scheduled/poll_mailbox.rb

```diff
@@ -96,12 +99,23 @@ def poll_pop3
           process_popmail(p)
         end
       end
-    rescue Net::OpenTimeout => e
-      mark_as_errored!
-      Discourse.handle_job_exception(e, error_context(@args, "Connecting to '#{SiteSetting.pop3_polling_host}' for polling emails."))
-    rescue Net::POPAuthenticationError => e
+    rescue Net::OpenTimeout, Net::POPAuthenticationError => e
+      if e.instance_of?(Net::OpenTimeout)
```

Comment:
Ah I'll revert it back to the previous style. I had to use the `instance_of?` pattern because I initially had to share a few similar method calls within the rescue block.